operator/api/v1alpha1/moduletemplate_types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ModuleTemplateSpec defines the desired state of ModuleTemplate
+type ModuleTemplateSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	Data map[string]string `json:"data,omitempty"`
```

Comment:
~~~~~~
Data TemplateData `json:"data,omitempty"`


type TemplateData struct {
      Kind string
      Group string
      Resource string
      Version string
      Settings map[string]interface{} // for component CR spec
}

Reply:
map[string]interface is not a valid spec unfortunately, i tried this but it didnt work out as I expected since it doesnt compile